lib/rendition.js

```diff
@@ -41,12 +74,74 @@ class Rendition {
         }
     }
 
+    exists() {
+        return fs.existsSync(this.path);
+    }
+
     size() {
-        return this.metadata[RenditionMetadata.REPO_SIZE];
+        return fs.statSync(this.path).size;
     }
 
-    sha1() {
-        return this.metadata[RenditionMetadata.REPO_SHA1];
+    async sha1() {
+        if (!this._sha1) {
+            this._sha1 = await fileHash(this.path, "sha1");
+        }
+        return this._sha1;
+    }
+
+    mimeType() {
+        // TODO: mimetype should be set by worker (this only looks at extension)
+        return mime.lookup(path.extname(this.path)) || DEFAULT_CONTENT_TYPE;
+    }
+
+    charset() {
+        // TODO: charset should be set by worker or determined by actually looking at the file
+        return mime.charset(path.extname(this.path));
+    }
+
+    contentType() {
+        // TODO: mimetype should be set by worker (this only looks at extension)
+        return mime.contentType(path.extname(this.path)) || DEFAULT_CONTENT_TYPE;
```

Comment:
That we  "guess" the mime type based on the filename extension here (this is all what `mime.contentType()` does) is actually a problem already as AEM still requests text renditions using the old `fmt=text-main`, so the rendition files are getting named `rendition0.text-main` and then it sees it as `application/octet-stream` since the mime library has no idea what `text-main` is.

As the comment says we need a way for workers to reliably communicate the actual mime type to the SDK for use here. The workers are the ones that know and it should not depend on that temporary rendition filename that the SDK comes up with.

This needs a separate channel, some options:
- static: set in `worker()` callback through some API, i.e. saying "this worker always generates mime type X"
- dynamic and/or shell script workers: store in a side car file (similar to `errors.json`)
- ...

Reply:
I would tend to go towards and API rather than using sidecar-files:
- files can "disappear" or fail to be written;
- I think it would be easier to reuse and extend the logic with an API/ callback in the SDK. So, if it is set by the worker, the worker, should always know which kind of file it generated... Otherwise how would it generate the proper file? The worker would be best placed to set the contenttype, then. Also, long long time ago, I started a toy library `node-file-mimetypes` (not public) to do that contentype recognition based on the file. I'm pretty sure we could adapt it to be used here. (The reason why we don't use it yet is the repo's test folder size. I have ideas to reduce it, but never implemented it)

Reply:
Yeah, but shell script workers have no API to call, exchanging a file is the only good mechanism. Environment variables don't work in the direction of child process to parent.

Reply:
created https://jira.corp.adobe.com/browse/NUI-607